libflux/flux-core/src/semantic/nodes.rs

```diff
@@ -1555,11 +1563,19 @@ impl MemberExpr {
     //
     fn infer(&mut self, infer: &mut InferState<'_, '_>) -> Result {
         self.object.infer(infer)?;
-        let t = self.object.type_of();
+        let t = self.object.type_of().apply(infer.sub);
+
+        if let Expression::Identifier(object) = &self.object {
+            if let Some(package_name) = infer.imports.get(&object.name) {
+                if let Some(property) = infer.importer.symbol(package_name, &self.property) {
+                    self.property = property;
```

Comment:
I am a little hazy in what's going on here, but is this saying that if we find a member expression like `foo.bar` and we have seen an `import "foo"` in the program that we annotate the property with the package `foo`?

Will this work right in places where we've shadowed the import package with an assignment? Looking back at the test cases you wrote in `semantic/tests.rs` I can see it works right but I'm unclear why.

Reply:
Yes, because `infer.imports`  is keyed by the `Symbol` for the `import` statement we only replace it if it comes from the import.